### PR TITLE
[15.0][FIX] contract: report translations & change structure

### PR DESCRIPTION
**Pull request by peluko00**
Problem to solve:
Go to Sales -> Orders -> Contracts and try to print a contract.
Normally when you print a contract the default language of that one bases on customer language but not works in that report.
Example:
Your default language is Spanish and your customer's default language is English
Try to print the report and you see that the report prints in Spanish and not in English.
I changed the structure because i'm base of the report of an invoice from the odoo's core structure(see more in the links below):
- https://github.com/odoo/odoo/blob/7f947871f09e0878470871522fdd7d39d4d04add/addons/account/views/report_invoice.xml#L429 
- https://github.com/odoo/odoo/blob/7f947871f09e0878470871522fdd7d39d4d04add/addons/account/views/report_invoice.xml#L4
I test that structure first and works perfect and I try to reproduce in that case and works fine.

[contract_report_failed.webm](https://github.com/OCA/contract/assets/91435961/eaf36fca-d2cd-497f-8f88-f514842b520d)

[contract_report_works.webm](https://github.com/OCA/contract/assets/91435961/4bfd35d4-8fce-4114-9936-04c1f0351d8c)

### CODE REVIEW

File: contract/data/mail_template.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
         <field name="partner_to">{{ object.partner_id.id }}</field>
         <field name="model_id" ref="model_contract_contract" />
         <field name="auto_delete" eval="True" />
-        <field name="report_template" ref="contract.report_contract" />
+        <field name="report_template" ref="contract.report_contract_action" />
```
Comment (pedrobaeza) [inline contract/data/mail_template.xml]: Avoid to rename things.

**Review by pedrobaeza — COMMENTED** [review]
In this kind of changes, screenshots of the before and after, and comments explaining what motivates the changes helps a lot to speed up reviews and avoid this to end stucked.

File: contract/data/mail_template.xml (hunk 1 above)
Comment (peluko00) [reply contract/data/mail_template.xml]: Fix it! Review please

**Comment by pedrobaeza** [comment]
What about the screenshots?

**Comment by peluko00** [comment]
Fixed Module report 

cc https://github.com/APSL 154015

@miquelalzanillas @lbarry-apsl @javierobcn @mpascuall please review

**Comment by peluko00** [comment]
> What about the screenshots?

Just upload two videos, one failing and the other with the current pr

File: contract/report/report_contract.xml [hunk 2]
```diff
@@ -1,209 +1,192 @@
...
     <template id="report_contract_document">
-        <t t-call="web.html_container">
-            <t t-foreach="docs" t-as="o">
-                <t t-call="web.external_layout">
-                    <t t-set="o" t-value="o.with_context(lang=o.partner_id.lang)" />
-                    <t t-set="address">
-                        <p id="partner_info">
-                            <strong>Partner:</strong>
+        <t t-call="web.external_layout">
+            <t t-set="o" t-value="o.with_context(lang=lang)" />
```
Comment (pedrobaeza) [inline contract/report/report_contract.xml:5]: Is this still needed?

Comment (peluko00) [reply contract/report/report_contract.xml:5]: Is necessary for translate partner information before starting the page with oe_structure. Can you review again please

Comment (peluko00) [reply contract/report/report_contract.xml:5]: Any update @pedrobaeza ?

**Review by pedrobaeza — APPROVED** [review]

**Comment by peluko00** [comment]
Ready for merge @pedrobaeza ?

**Comment by pedrobaeza** [comment]
Second review missing

**Review by lbarry-apsl — APPROVED** [review]
It is working tested in Runboat

**Comment by peluko00** [comment]
> Second review missing

Second review done

**Review by miquelalzanillas — APPROVED** [review]

**Pull request merged**